front/types/assistant/conversation.ts

```diff
@@ -0,0 +1,119 @@
+import { ModelId } from "@app/lib/models";
+
+import { UserProviderType, UserType } from "../user";
+
+/**
+ * Mentions
+ */
+
+export type AssistantAgentMention = {
+  assistantId: string;
+};
+
+export type AssistantUserMention = {
+  provider: UserProviderType;
```

Comment:
Why don't we point to a front.users.id field?

Reply:
So far ModelId is only used as a convenience on some object for us, and the invariant is that we should be able to switch data store transparently. The real business logic pointer to a user is its provider and providerId, we could add a shortId to circumvent that, but I'm quite bearish on relying entirely on the SQL ID.

Reply:
Ok - I would be curious to know why you don't want to rely on this SQL id but we can have that conversation later.
I am fine with this rational.